src/pudl/output/ferc1.py

```diff
@@ -1249,6 +1259,64 @@ def _aggregatable_dimension_tags(
     return tags_df[tags_df[aggregatable_col] != "total"]
 
 
+def make_correction_tags(
+    tags_all: pd.DataFrame, calc_components: pd.DataFrame
+) -> pd.DataFrame:
+    """Make tags for correction records.
+
+    We need to check to see if any of the tags in each of the calculated
+    parent factoids are the same for all of their child components. So in this
+    function, we're going to merge on the tags to the children then groupby the
+    parents. For each tag, see if the childrens'tags contains only one unique value.
+    If so grab the tag to associate with the correction record of the parent. If not,
+    no tag will be associated with the record.
+    """
```

Comment:
this is my new attempt at adding tags to parent facts when all of their children share the same tag. BUT i think this could probably be done more switfly and more completely using the networkx graphs. if all ancestors of a node share tag X, give all ancestor nodes tag X.